compiler/cli/cli-runner/src/org/jetbrains/kotlin/runner/runners.kt

```diff
@@ -37,23 +37,23 @@ abstract class AbstractRunner : Runner {
 
         val mainClass = try {
             classLoader.loadClass(className)
-        }
-        catch (e: ClassNotFoundException) {
+        } catch (e: ClassNotFoundException) {
+            throw RunnerException("could not find or load main class $className")
+        } catch (e: NoClassDefFoundError) {
             throw RunnerException("could not find or load main class $className")
         }
```

Comment:
This change needed here to print an error message instead of the exception stack. It is related to [KT-46171](https://youtrack.jetbrains.com/issue/KT-46171), I created that issue earlier today.

Reply:
And adjusted the format of  file `runners.kt` , hope it is okay.

Reply:
I'd prefer not to catch this exception because doing so hides the JVM error message which contains useful information (`temp/HelloKt (wrong name: HelloKt)`) which can help the user in fixing the command line string.

Alternatively, we can add `cause: Throwable?` to `RunnerException`, and keep the underlying cause in certain cases like this one, and then display e.g. the first line (exception class name + message) to the user. This is basically what the `java` executable does, both when trying to run a non-existent class, and when trying to run a class with an incorrect name, such as in KT-46171. But that might require a bit more work and tests, so maybe it's not feasible in this review.